apps/site/lib/site/application.ex

```diff
@@ -39,6 +39,7 @@ defmodule Site.Application do
         if Application.get_env(:elixir, :start_data_processes) do
           [
             {Site.Stream.Vehicles, name: Site.Stream.Vehicles},
+            {Site.Stream.Vehicles, name: Site.Stream.Vehicles, topic: "vehicles-v2"},
```

Comment:
question: Do we need the second GenServer? Having separate channels makes sense to me, but could `VehicleMapMarkerChannel` and `VehicleChannel` both subscribe to the same Vehicles pub-sub stream? (I might be confusing myself about the different stages here…)

If that doesn't make sense, a more descriptive topic might be helpful. (And maybe topics for both?)

Reply:
Yeah good point! Instead I'll have a single gen server take a list of topics to broadcast to.

Reply:
That certainly works too, but could there even be a single topic? Are the two topics broadcasting the same data? (Maybe they aren't and I just missed something.)

Reply:
I think I see what you mean! This is pushing the boundaries of my current phoenix channel knowledge, but maybe something like[ subscribing to external topics](https://hexdocs.pm/phoenix/Phoenix.Channel.html#module-subscribing-to-external-topics) would support that. Though I'll have to think more about this piece

> Note: the caller must be responsible for preventing duplicate subscriptions.

Reply:
Maybe this is me misunderstanding and confusing things. I thought that `Site.Stream.Vehicles` was publishing over Phoenix PubSub to `VehicleMapMarkerChannel`, and then `VehicleMapMarkerChannel` is publishing over the socket channel to the front-end client. Do I have that correct? If so, I was wondering if it was possible to leave the channels as they are, but feed them both from the same Phoenix PubSub feed, like:

```
Site.Stream.Vehicles  ->  VehicleMapMarkerChannel  ->  "vehicles:*"
		      ↳  VehicleChannel  ->  "vehicles-v2:*"
```

But maybe that doesn't make sense in reality.

Reply:
I think I see what you mean, but perhaps the current limitation is that `Site.Stream.Vehicles` isn't using Phoenix `PubSub.broadcast/3`- it is using `Endpoint.broadcast/3` to broadcast directly to the "vehicles" channel topic. 
With that, am stuck at how to execute on the ` ↳` piece of your diagram diagram.

I think to do what you describe without modifying `Site.Stream.Vehicles`, `VehicleChannel` would have to also subscribe to the "vehicles" topic, but push messages to clients subscribed to "vehicles-v2". One way to achieve that seems to be 'subscribing to external topics', but that seems like it could result in duplicate subscriptions in the case where a single client is subscribed to both the `VehicleMapMarkerChannel` and `VehicleChannel`. 

Alternatively, I could change `Site.Stream.Vehicles` to broadcast a to an internal topic not associated with a channel, and have both channels subscribe to that topic on join. Though that feels like it could be more substantial change

Reply:
Ah, I see. Sorry, I was definitely confused about that. In that case, I think I would just revoke the original suggestion in the first place. Sorry for all the confusion!

Reply:
Sorry for not clarifying that sooner, I was still working through the details of it all myself! Do you think it would still be worth consolidating to use a single GenServer which takes a list of topics?

Reply:
Up to you, but I might say maybe it doesn't seem worthwhile. At first, I thought the two modules might be doing the same thing, but now knowing that they aren't, I think one per channel seems totally reasonable and maybe even easier to understand.